### DISCUSSION TITLE
Web3 Connection Issue.

### DISCUSSION THREAD

**Discussion by naji0329** [community/community#19950] [Projects and Issues] [answered]
HI.

Now I am trying to build contract with abi and contract address.

When I make new web3, I used this line.
     const provider = new Web3.providers.HttpProvider(rpcUrls);
![IMG-20220626-WA0024 - Copy - Copy](https://user-images.githubusercontent.com/94976661/177072256-424beb0d-e64e-4a5b-8588-8ec1d3441728.jpg)

     const web3 = new Web3(provider);

But when I call contract's approve function, it is not working. Read function is working well, but Write function is not working.
If you know the reason, Please let me know.
Thanks.

**Comment by miracle0315** [comment, marked as answer]
I saw your error.
I think you can solve this issue with below code.

 const web3 = new Web3(window.ethereum)
;
